internal/internal_worker.go

```diff
@@ -78,14 +78,15 @@ type (
 	// And worker is mapped 1:1 with task list. If the user want's to poll multiple
 	// task list names they might have to manage 'n' workers for 'n' task lists.
 	workflowWorker struct {
-		executionParameters workerExecutionParameters
-		workflowService     workflowserviceclient.Interface
-		domain              string
-		poller              taskPoller // taskPoller to poll and process the tasks.
-		worker              *baseWorker
-		localActivityWorker *baseWorker
-		identity            string
-		stopC               chan struct{}
+		executionParameters             workerExecutionParameters
+		workflowService                 workflowserviceclient.Interface
+		domain                          string
+		poller                          taskPoller // taskPoller to poll and process the tasks.
+		worker                          *baseWorker
+		localActivityWorker             *baseWorker
+		locallyDispatchedActivityWorker *baseWorker
```

Comment:
We don't need a new worker for this. Can reuse the existing activity worker.

Reply:
Discussed offline: lets compare the implementations complexity.

Reply:
Decided to create the activity worker outside of workflow worker. Later we will refactor the code such that the taskPoller interface returns a channel for tasks, and move the for loop into the task poller implementation.